server/stream.go

```diff
@@ -1291,20 +1291,10 @@ func (s *Server) checkStreamCfg(config *StreamConfig, acc *Account) (StreamConfi
 	if cfg.RePublish != nil {
 		// Check to make sure source is a valid subset of the subjects we have.
 		// Also make sure it does not form a cycle.
-		var srcValid bool
 		// Empty same as all.
 		if cfg.RePublish.Source == _EMPTY_ {
 			cfg.RePublish.Source = fwcs
 		}
-		for _, subj := range cfg.Subjects {
```

Comment:
Anyway to tell if their config will never publish any messages?

Reply:
Fair, I wonder though if we do best effort to some degree? Or you think leave without checks/

Reply:
I would leave without checks, in case someone sets republish with a filter that doesn't match any messages in the stream it's pretty easy now to check that using `nats stream subjects` which will show you exactly what is in the stream not just trying to check with sourcing, sourcing chains, subject transforms, etc...